[Bug][T2]: `gnmi/test_gnmi_configdb.py` fails on modular systems if supervisor is selected `duthost`

### Issue Description

`gnmi/test_gnmi_configdb.py` has a couple of tests which rely on ports existing on the `duthost`
TEST1 - `test_gnmi_configdb_incremental_01`
```
  File "/data/tests/gnmi/test_gnmi_configdb.py", line 57, in test_gnmi_configdb_incremental_01
    assert interface is not None, "Invalid interface"
AssertionError: Invalid interface
```
It's fetching the interface from `CONFIG_DB` of the `duthost`:
https://github.com/sonic-net/sonic-mgmt/blob/master/tests/gnmi/test_gnmi_configdb.py#L48
```
def get_interface_status(duthost, field, interface='Ethernet0'):
    cmds = 'sonic-db-cli CONFIG_DB hget "PORT|{}" {}'.format(interface, field)
```

TEST2 - `test_gnmi_configdb_full_01`
```
  File "/data/tests/gnmi/test_gnmi_configdb.py", line 117, in test_gnmi_configdb_full_01
    assert "PORT" in dic, "Failed to read running config"
AssertionError: Failed to read running config
```
It's getting the port data from `sonic-cfggen` on the `duthost`:
https://github.com/sonic-net/sonic-mgmt/blob/master/tests/gnmi/test_gnmi_configdb.py#L227
```
output = duthost.shell("sonic-cfggen -d --print-data")
assert (not output['rc']), "No output"
dic = json.loads(output["stdout"])
assert "PORT" in dic, "Failed to read running config"
```

The problem is the fixture used by these tests is `rand_one_dut_hostname` which includes the supervisor which won't have any ports on it in modular systems.

We could update the tests to instead use `rand_one_dut_front_end_hostname` but this will require updating all the `autouse=True` fixtures in `gnmi/conftest` and will therefor impact the other `gnmi` tests like `gnmi/test_gnmi.py` and `gnmi/test_gnmi_appldb.py`


### Results you see

`gnmi/test_gnmi_configdb.py` is being run on supervisor `duthost` but the test requires PORTs on the `duthost`

### Results you expected to see

`gnmi/test_gnmi_configdb.py` should likely only run against the LC `duthosts`

### Is it platform specific

generic

### Relevant log output

_No response_

### Output of `show version`

_No response_

### Attach files (if any)

_No response_

--- a/tests/gnmi/test_gnmi_configdb.py
+++ b/tests/gnmi/test_gnmi_configdb.py
@@ -69,6 +69,8 @@ def test_gnmi_configdb_incremental_01(duthosts, rand_one_dut_hostname, ptfhost):
     Toggle interface admin status
     '''
     duthost = duthosts[rand_one_dut_hostname]
+    if duthost.is_supervisor_node():
+        pytest.skip("gnmi test relies on port data not present on supervisor card '%s'" % rand_one_dut_hostname)
     file_name = "port.txt"
     interface = get_first_interface(duthost)
     assert interface is not None, "Invalid interface"
@@ -236,6 +238,8 @@ def test_gnmi_configdb_full_01(duthosts, rand_one_dut_hostname, ptfhost):
     Toggle interface admin status
     '''
     duthost = duthosts[rand_one_dut_hostname]
+    if duthost.is_supervisor_node():
+        pytest.skip("gnmi test relies on port data not present on supervisor card '%s'" % rand_one_dut_hostname)
     interface = get_first_interface(duthost)
     assert interface is not None, "Invalid interface"
 

